tests/Utils/Models/User.php

```diff
@@ -59,8 +59,8 @@ public function roles(): BelongsToMany
 
     public function scopeCompanyName(Builder $query, array $args): Builder
     {
-        return $query->whereHas('company', function (Builder $q) use ($args): void {
-            $q->where('name', $args['company']);
-        });
+        return $query
+            ->join('companies', 'users.company_id', '=', 'companies.id')
```

Comment:
Why would you want to use this over the way it was written initially?

Reply:
The original scope was using `whereHas()`, which creates an inner query and becomes immune to the bug I'm trying to fix. In our code base, we have cases where scopes join further tables to the builder, which is as far as I know, valid Eloquent usage. I changed the code in order to trigger the buggy behaviour in integration tests.

Reply:
I think that the `whereHas` is idiomatic Eloquent style, the Laravel docs only show that https://laravel.com/docs/5.8/eloquent-relationships#querying-relationship-existence

Manually joining seems unnecessary and having to add on the table name to every column seems extremely tedious.